### ISSUE TITLE
Unable to set password/enable E2EE for rooms.

### ISSUE THREAD

**Issue by Slips-PC** [jitsi/jitsi-meet#11620]
### Description:

When starting a new room, clicking on Security Options in the overflow menu provides me a prompt that says "Password: none" with no visible way to change it.

### Steps to reproduce:

1. Connect to jitsi.slipfox.xyz
2. Join a meeting with no users, thus setting yourself as the host.
3. Click on Security Options in the overflow menu.

### Expected behavior:

You should be provided with some options relating to things such as lobby and password.

### Actual behavior:

No options are provided.
![image](https://user-images.githubusercontent.com/59989951/172032687-aaef830a-6762-4582-a9d6-7927ba580d6e.png)

### Server information:

- Jitsi Meet version: 2.0.7287-1
- Operating System: Debian 11

### Client information:

- Browser / app version: Firefox 100.0.2
- Operating System: Gentoo Linux

**Comment by saghul** [comment]
That means you are not a moderator and such cannot turn those options on.

If you try meet.jit.si, where every user is a moderator you'll have those options available.

**Comment by Slips-PC** [comment]
How do I mark users as moderators?

**Comment by Slips-PC** [comment]
To be clear, this is an empty room *I* created. So i should've been already marked as the host, right?

**Comment by damencho** [comment]
Not necessary, depends on your config. Share your jicofo and prosody config.

**Comment by Slips-PC** [comment]
Alright.
Prosody:
```
plugin_paths = { "/usr/share/jitsi-meet/prosody-plugins/" }

-- domain mapper options, must at least have domain base set to use the mapper
muc_mapper_domain_base = "jitsi.slipfox.xyz";

external_service_secret = "UNjyhxUU4lsv32LL";
external_services = {
     { type = "stun", host = "jitsi.slipfox.xyz", port = 3478 },
     { type = "turn", host = "jitsi.slipfox.xyz", port = 3478, transport = "udp", secret = true, ttl = 86400, algorithm = "turn" },
     { type = "turns", host = "jitsi.slipfox.xyz", port = 5349, transport = "tcp", secret = true, ttl = 86400, algorithm = "turn" }
};

cross_domain_bosh = false;
consider_bosh_secure = true;
-- https_ports = { }; -- Remove this line to prevent listening on port 5284

-- by default prosody 0.12 sends cors headers, if you want to disable it uncomment the following (the config is available on 0.12.1)
--http_cors_override = {
--    bosh = {
--        enabled = false;
--    };
--    websocket = {
--        enabled = false;
--    };
--}

-- https://ssl-config.mozilla.org/#server=haproxy&version=2.1&config=intermediate&openssl=1.1.0g&guideline=5.4
ssl = {
    protocol = "tlsv1_2+";
    ciphers = "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384"
}

unlimited_jids = {
    "focus@auth.jitsi.slipfox.xyz",
    "jvb@auth.jitsi.slipfox.xyz"
}

VirtualHost "jitsi.slipfox.xyz"
    -- enabled = false -- Remove this line to enable this host
    authentication = "anonymous"
    -- Properties below are modified by jitsi-meet-tokens package config
    -- and authentication above is switched to "token"
    --app_id="example_app_id"
    --app_secret="example_app_secret"
    -- Assign this host a certificate for TLS, otherwise it would use the one
    -- set in the global section (if any).
    -- Note that old-style SSL on port 5223 only supports one certificate, and will always
    -- use the global one.
    ssl = {
        key = "/etc/prosody/certs/jitsi.slipfox.xyz.key";
        certificate = "/etc/prosody/certs/jitsi.slipfox.xyz.crt";
    }
    av_moderation_component = "avmoderation.jitsi.slipfox.xyz"
    speakerstats_component = "speakerstats.jitsi.slipfox.xyz"
    conference_duration_component = "conferenceduration.jitsi.slipfox.xyz"
    -- we need bosh
    modules_enabled = {
        "bosh";
        "pubsub";
        "ping"; -- Enable mod_ping
        "speakerstats";
        "external_services";
        "conference_duration";
        "muc_lobby_rooms";
        "muc_breakout_rooms";
        "av_moderation";
    }
    c2s_require_encryption = false
    lobby_muc = "lobby.jitsi.slipfox.xyz"
    breakout_rooms_muc = "breakout.jitsi.slipfox.xyz"
    main_muc = "conference.jitsi.slipfox.xyz"
    -- muc_lobby_whitelist = { "recorder.jitsi.slipfox.xyz" } -- Here we can whitelist jibri to enter lobby enabled rooms

Component "conference.jitsi.slipfox.xyz" "muc"
    restrict_room_creation = true
    storage = "memory"
    modules_enabled = {
        "muc_meeting_id";
        "muc_domain_mapper";
        "polls";
        --"token_verification";
        "muc_rate_limit";
    }
    admins = { "focus@auth.jitsi.slipfox.xyz" }
    muc_room_locking = false
    muc_room_default_public_jids = true

Component "breakout.jitsi.slipfox.xyz" "muc"
    restrict_room_creation = true
    storage = "memory"
    modules_enabled = {
        "muc_meeting_id";
        "muc_domain_mapper";
        --"token_verification";
        "muc_rate_limit";
        "polls";
    }
    admins = { "focus@auth.jitsi.slipfox.xyz" }
    muc_room_locking = false
    muc_room_default_public_jids = true

-- internal muc component
Component "internal.auth.jitsi.slipfox.xyz" "muc"
    storage = "memory"
    modules_enabled = {
        "ping";
    }
    admins = { "focus@auth.jitsi.slipfox.xyz", "jvb@auth.jitsi.slipfox.xyz" }
    muc_room_locking = false
    muc_room_default_public_jids = true

VirtualHost "auth.jitsi.slipfox.xyz"
    ssl = {
        key = "/etc/prosody/certs/auth.jitsi.slipfox.xyz.key";
        certificate = "/etc/prosody/certs/auth.jitsi.slipfox.xyz.crt";
    }
    modules_enabled = {
        "limits_exception";
    }
    authentication = "internal_hashed"

-- Proxy to jicofo's user JID, so that it doesn't have to register as a component.
Component "focus.jitsi.slipfox.xyz" "client_proxy"
    target_address = "focus@auth.jitsi.slipfox.xyz"

Component "speakerstats.jitsi.slipfox.xyz" "speakerstats_component"
    muc_component = "conference.jitsi.slipfox.xyz"

Component "conferenceduration.jitsi.slipfox.xyz" "conference_duration_component"
    muc_component = "conference.jitsi.slipfox.xyz"

Component "avmoderation.jitsi.slipfox.xyz" "av_moderation_component"
    muc_component = "conference.jitsi.slipfox.xyz"

Component "lobby.jitsi.slipfox.xyz" "muc"
    storage = "memory"
    restrict_room_creation = true
    muc_room_locking = false
    muc_room_default_public_jids = true
    modules_enabled = {
        "muc_rate_limit";
        "polls";
    }

-- Enabled dial-in for JaaS customers
-- Note: make sure you have the following packages installed: lua-basexx, liblua5.3-dev, libssl-dev, luarocks
-- and execute $ sudo luarocks install luajwtjitsi 3.0-0
VirtualHost "jigasi.meet.jitsi"
    enabled = false -- JaaS customers remove this line
    modules_enabled = {
      "ping";
      "bosh";
    }
    authentication = "token"
    app_id = "jitsi";
    asap_key_server = "https://jaas-public-keys.jitsi.net/jitsi-components/prod-8x8"
    asap_accepted_issuers = { "jaas-components" }
    asap_accepted_audiences = { "jigasi.jitsi.slipfox.xyz" }
```
jicofo
```
# Jicofo HOCON configuration. See reference.conf in /usr/share/jicofo/jicofo.jar for
#available options, syntax, and default values.
jicofo {
  xmpp: {
    client: {
      client-proxy: focus.jitsi.slipfox.xyz
    }
    trusted-domains: [ "recorder.jitsi.slipfox.xyz" ]
  }
  bridge: {
    brewery-jid: "JvbBrewery@internal.auth.jitsi.slipfox.xyz"
  }
}
```

**Comment by damencho** [comment]
Looks good. Do you see any errors in prosody logs, or jicofo logs?
By default, jicofo will make moderator the first participant to join a room.

**Comment by Slips-PC** [comment]
I'm seeing some health check NullPointerExceptions in jicofo (see below), and nothing from prosody.
```
Jicofo 2022-06-05 22:46:09.487 SEVERE: [138] JvbDoctor$AbstractHealthCheckTask.run#271: Error when doing health-check on: Bridge[jid=jvbbrewery@internal.auth.jitsi.slipfox.xyz/eb1fe2b1-e2a3-4fd0-b69b-0a8ee0e1aec9, version=2.1.681-g3544ed05, relayId=null, region=null, stress=0.00]
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at org.jitsi.jicofo.bridge.JvbDoctor.getConnection(JvbDoctor.java:76)
	at org.jitsi.jicofo.bridge.JvbDoctor$HealthCheckTask.doHealthCheck(JvbDoctor.java:156)
	at org.jitsi.jicofo.bridge.JvbDoctor$AbstractHealthCheckTask.run(JvbDoctor.java:260)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by damencho** [comment]
Hum, this is suspicious. What about if you clean jicofo log and restart jicofo? Any errors then?

**Comment by Slips-PC** [comment]
`rm`ing the logfile and restarting the jicofo systemd unit does nothing. the same errors are printed.

**Comment by Slips-PC** [comment]
Update: I did also try using the jitsi meet official instance to see if my browser worked with E2EE on jitsi (it doesn't), so i tried the jitsi meet client and discovered that my instance will not connect to jitsi meet at all. So that is interesting.

**Comment by saghul** [comment]
Any chance you have disabled WASM in your browser? That would explain why E2EE is missing also on meet.jit.si.

**Comment by Slips-PC** [comment]
Possible, but unlikely. I was testing it with the jitsi client as well, where i have confirmed E2EE to work, so that *should* have worked, but as mentioned in my previous comment i can't even connect to my instance at all on the official Electron client.

**Comment by Slips-PC** [comment]
So, after a bit of tweaking, I can now access the "Join Room" prompt on the Electron client where you can enable audio/video before actually joining, but the Join Room button does nothing. No logs, no errors, nothing. This is truly baffling.

**Comment by damencho** [comment]
Check browser console logs, any errors there?

What do you see when you access https://yourdeploymemt.com/http-bind ?

**Comment by Slips-PC** [comment]
/http-bind provides a message about a BOSH endpoint.

As for browser console logs, I see one warning about stream resume being enabled but websockets being disabled, and a handful of deprecation warnings, but nothing else.

**Comment by damencho** [comment]
Clean jicofo logs, restart jicofo, and upload the logs here.

**Comment by Slips-PC** [comment]
Wow, that was incredibly stupid of me. I went to go delete `/var/log/jitsi/jicofo.log` and for some reason typed in `/etc/jitsi/jicofo/jicofo.conf`. Whoops.
I did figure out what it was though. There was *one* error I missed in my readings: A port bind error. Apparently I already have a service running on :8888 that i forgot about. Borrowed a config from a friend who also runs jitsi, and set the port to a different one, and it works perfectly now.
Thank you for the assistance, sorry for the bother.
